app/controllers/supermarkets_controller.rb

```diff
@@ -0,0 +1,6 @@
+class SupermarketsController < ApplicationController
+  def show
+    @supermarket = Supermarket.find(params[:id])
+    @customers = @supermarket.find_customers
```

Comment:
I would probably just call `@supermarket.find_customers` in your view and not store it here in this instance variable.